src/plugin-client.ts

```diff
@@ -9,28 +9,58 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
-'use strict';
+import {AddonManagerProxy} from './addon-manager-proxy';
+import {MessageType} from './constants';
+import {Deferred} from './deferred';
+import {EventEmitter} from 'events';
+import {IpcSocket} from './ipc';
+import WebSocket from 'ws';
 
-const AddonManagerProxy = require('./addon-manager-proxy');
-const {MessageType} = require('./constants');
-const Deferred = require('./deferred');
-const EventEmitter = require('events');
-const IpcSocket = require('./ipc');
+export class PluginClient extends EventEmitter {
+  private pluginId: string;
 
-class PluginClient extends EventEmitter {
-  constructor(pluginId, {verbose} = {}) {
+  private verbose: boolean;
+
+  private deferredReply?: Deferred<AddonManagerProxy, void> | null;
+
+  private logPrefix: string;
+
+  private gatewayVersion?: string;
+
+  private userProfile: any;
+
+  private preferences: any;
+
+  private addonManager?: AddonManagerProxy;
+
+  private ipcSocket?: IpcSocket;
+
+  private ws?: WebSocket;
+
+  constructor(pluginId: string, {verbose}: Record<string, unknown> = {}) {
     super();
     this.pluginId = pluginId;
-    this.verbose = verbose;
-    this.deferredReply = null;
+    this.verbose = !!verbose;
     this.logPrefix = `PluginClient(${this.pluginId}):`;
   }
 
-  onMsg(msg) {
+  getGatewayVersion(): string | undefined {
+    return this.gatewayVersion;
+  }
+
+  getUserProfile() {
+    return this.userProfile;
+  }
+
+  getPreferences() {
```

Comment:
nitpick: no return type

Reply:
I intentionally left them open because I wanted to generate the types first.
This should be fixed now.